### DISCUSSION TITLE
Creating Auth Context From public.users Instead of auth.users?

### DISCUSSION THREAD

**Discussion by hjaber** [supabase/supabase#3667] [Questions]
What are your favorite ways of storing the current user when using a public.users table instead of auth.users as the main user object? Motivation being, more flexibility with things like usernames and/or posts.

It seems a subscription might be the best approach to keep up with any changes, but not sure how performant that would be.  Using the below function with Nextjs and storing in context but wondering if there is a better way.

There are great docs and many examples when using auth.users but not too many when using a public.user table.

Thanks!

```
  const getUser = async () => {
    const supabaseUser = await supabase.auth.user();
    const { data: currentPublicUser } = await supabase
      .from("users")
      .select("*")
      .eq("id", supabaseUser?.id);
  };
```

**Comment by GaryAustin1** [comment]
I have a public.user table with nickname, type of account, various settings and selected options, favorites, etc.  Some of these are set up by request at start of interaction with user and others over time.  I read this at "start up" every time.

Users can have multiple devices so I created a separate status table with per user entry with various flags for changes to public.user entry or last note written, etc.  I use realtime subscription on this table to get flagged for changes and then do RLS reads from the public user table.  Right now realtime does not work with RLS so best not to have a subscription to tables with sensitive data.  Status is updated by update triggers on the user table or other tables of concern.

**Reply by hjaber** [reply]
> I have a public.user table with nickname, type of account, various settings and selected options, favorites, etc. Some of these are set up by request at start of interaction with user and others over time. I read this at "start up" every time.
> 
> Users can have multiple devices so I created a separate status table with per user entry with various flags for changes to public.user entry or last note written, etc. I use realtime subscription on this table to get flagged for changes and then do RLS reads from the public user table. Right now realtime does not work with RLS so best not to have a subscription to tables with sensitive data. Status is updated by update triggers on the user table or other tables of concern.

Ah this is very clever and I like this workflow of both RLS and realtime subscription, my attempt to summarize:

Tables:
auth.users 
public.users read on site render, RLS, triggered by updates to auth.users, keeps user information
public.userStatus realtime, no RLS, triggered by updates to public.users, records status of public.users

For your workflow, let's say a user marks something as a favorite or writes a new note, is the order: 

1. public.users update triggers public.userStatus 
2. public.userStatus update is subscribed to on frontend, which then causes site to re-read the public.users auth object on frontend?

Thanks!

**Reply by GaryAustin1** [reply]
Yeah. But I'm using Status table for monitoring multiple tables not just public.users.  Status is basically a couple of timestamp flags for different tables' user specific info being updated so they change when updated and can be compared in the realtime payload to see what changed and take action (reload if needed).
